libc/string_errsig.c

```diff
@@ -0,0 +1,312 @@
+		char buf[MAX_LEN_STRING] = { 0 };
+
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i], buf, sizeof(buf)));
+
+		if (i != 0) {
+			TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i - 1], prev_buf, sizeof(prev_buf)));
+			TEST_ASSERT_NOT_EQUAL_INT(0, strcmp(buf, prev_buf));
+		}
+
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsign, strerror_r_zero)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	char buf[MAX_LEN_STRING];
+	errno = 0;
+	TEST_ASSERT_EQUAL_INT(0, strerror_r(0, buf, sizeof(buf)));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+#endif
+}
+
+
+TEST(string_errsign, strerror_r_invaild)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	const int values[] = { INT_MIN, INT_MIN / 2, INT_MIN / 4, -1024, -256, 256, 1024, INT_MAX / 4, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		char buf[MAX_LEN_STRING] = { 0 };
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(EINVAL, strerror_r(values[i], buf, sizeof(buf)));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsign, strerror_r_overflow)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	char buf[1];
+	char buf2[MAX_LEN_STRING];
+	for (int i = 0; i < error_codes_len; i++) {
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(ERANGE, strerror_r(error_codes[i], buf, sizeof(buf)));
+		TEST_ASSERT_EQUAL_INT(ERANGE, strerror_r(error_codes[i], buf2, 2));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+#endif
+}
+
+
+TEST(string_errsign, strsignal_basic)
+{
+	for (int i = 0; i < signal_codes_len; i++) {
+		errno = 0;
+		TEST_ASSERT_NOT_NULL(strsignal(signal_codes[i]));
+
+		if (i != 0) {
+			TEST_ASSERT_NOT_NULL(strsignal(signal_codes[i - 1]));
+			TEST_ASSERT_NOT_EQUAL_INT(0, strcmp(strsignal(signal_codes[i - 1]), strsignal(signal_codes[i])));
+		}
+
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(string_errsign, strsignal_invaild)
+{
+	const int values[] = { INT_MIN, INT_MIN / 2, INT_MIN / 4, -1024, -256, -128, -64, -1, 0, 128, 256, 1024, INT_MAX / 4, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		errno = 0;
+		TEST_ASSERT_NOT_NULL(strsignal(signal_codes[i]));
+		TEST_ASSERT_NOT_NULL(strstr(strsignal(values[i]), "Unknown signal"));
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(string_errsign, strsignal_real_time)
+{
+/* Disabled because of #687 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/687 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#687 issue");
+#else
+	TEST_ASSERT_NOT_NULL(strsignal(SIGRTMIN));
+	TEST_ASSERT_NOT_NULL(strstr(strsignal(SIGRTMIN), "0"));
```

Comment:
As we decided to not check the exact content of an error message I would suggest checking whether it's not NULL.

Alternatively we could leave this approach, but then it would be good to check those numbers dynamically, so not 30, but `SIGRTMAX`, because it's not specified that SIGRTMAX should equal 30. Still the more flexible solution with asserting not NULL seems to be better. Above that, we could check all the possible `RT` values (SIGRTMIN..SIGRTMAX)